### ISSUE TITLE
No rule to make target '/usr/local/lib/libneo4j-client.a'

### ISSUE THREAD

**Issue by eserte** [majensen/perlbolt#28]
On my Debian smoker systems (with libneo4j-client-dev package installed) build+test fails:
```
Output from './Build test':

Use of uninitialized value $liba in pattern match (m//) at Builder.pm line 30.
Building Neo4j-Bolt
validate Stage
Starting Build Preprocess Stage
get_maps Stage
Finished Build Preprocess Stage
...
  Starting "make" Stage
make: *** No rule to make target '/usr/local/lib/libneo4j-client.a', needed by 'subdirs'.  Stop.
...
```
The library exists, but in another location: /usr/lib/x86_64-linux-gnu/libneo4j-client.a

**Comment by majensen** [comment]
Thanks @eserte - #29 won't fix this on your smokers, but #30 will. If you want to have a look that would be very helpful. 
Can you tell me what happens when you automatically run Build.PL? I tried to exit with 0 before creating the build script when the dep (libneo4j-client.a) is not found or specified. That was v0.11, but I see that ./Build still gets created in the smoke test on that version.
Much appreciated (as usual)

**Comment by eserte** [comment]
`Build` is correctly created --- as I said, libneo4j-client.a exists on my Debian:10 smoker, so `exit(0)` is never called.

**Comment by majensen** [comment]
Ok, now I get it.

**Comment by majensen** [comment]
will see if v0.12 fixes

**Comment by johannessen** [comment]
[CPAN Testers Reports: FAIL Neo4j-Bolt-0.12 5.30.1 GNU/Linux](http://www.cpantesters.org/cpan/report/f7e3fa6e-5b00-11ea-b803-88131f24ea8f):
````
LD_RUN_PATH="/usr/lib/x86_64-linux-gnu" cc  -shared -O2 -L/usr/local/lib -fstack-protector-strong  TypeHandlersC.o  -o blib/arch/auto/Neo4j/Bolt/TypeHandlersC/TypeHandlersC.so /usr/lib/x86_64-linux-gnu/libneo4j-client.a \
   -lneo4j-client   \
  
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libneo4j-client.a(libneo4j_client_la-values.o): relocation R_X86_64_PC32 against symbol `neo4j_ntostring' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libneo4j-client.a(libneo4j_client_la-print.o): relocation R_X86_64_PC32 against symbol `NEO4J_STRING' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libneo4j-client.a(libneo4j_client_la-serialization.o): relocation R_X86_64_PC32 against symbol `NEO4J_NULL' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libneo4j-client.a(libneo4j_client_la-util.o): relocation R_X86_64_TPOFF32 against `index.5407' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libneo4j-client.a(libneo4j_client_la-iostream.o): relocation R_X86_64_TPOFF32 against `_diov.5984' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: nonrepresentable section on output
collect2: error: ld returned 1 exit status
make: *** [Makefile:485: blib/arch/auto/Neo4j/Bolt/TypeHandlersC/TypeHandlersC.so] Error 1
````

What’s weird is that I tried to reproduce this on a Debian system like Slaven’s, but I get *different* errors. I’ll try to look into it.

**Comment by johannessen** [comment]
Doh, my Debian system is “oldstable”, which only provides libneo4j-client version 1.2.1. No wonder I get weird errors here.

But that doesn’t explain the errors Slaven gets (quoted above).

**Comment by majensen** [comment]
those refer to how the libneo4j-client lib was created -- probably there is a workaround, but the "recompile with -fPIC" on the client side would be better. :|

**Comment by majensen** [comment]
but the patch found the lib!

**Comment by johannessen** [comment]
Yeah, but is it useful to find a lib that can’t be linked? :)

Not sure what the next step should be here.

**Comment by majensen** [comment]
🤔well I can kludge in something that can look for that error, just to keep the test from failing. Maybe our build could be more flexible to allow static libs like that. If that a standard distribution for libneo4j-client, then  I suppose we want to try to support it.

**Comment by johannessen** [comment]
Are we absolutely sure this error message isn’t referring to us? Should we perhaps try adding add that flag ourselves, just in case?

I mean, I do agree that the error message sounds like it’s the neo4j-client lib, but we are *also* creating a shared library here, right? Performance considerations aside, I guess it wouldn’t hurt for Neo4j::Bolt to be compiled as position independent.

**Comment by majensen** [comment]
I believe this is now fixed with the release of v0.20. This incorporates [Neo4j::Client](https://metacpan.org/pod/Neo4j::Client), which builds and encapsulates a local version of the client library.
So far, seems to be passing on @eserte 's linux servers [e.g.](http://www.cpantesters.org/cpan/report/9c45ed08-a79e-11ea-bc57-fb801f24ea8f).
